src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2481,25 +2553,22 @@ private static ImmutableArray<FieldOrPropertyInitializer> ToReadonlyAndFree(Arra
                     }
                 }
 
-                group.Free();
                 return builder.ToImmutableAndFree();
             }
 
-            public void Free()
-            {
-                NonTypeNonIndexerMembers.Free();
-                StaticInitializers.Free();
-                InstanceInitializers.Free();
-                IndexerDeclarations.Free();
-            }
-
             internal void AddOrWrapTupleMembers(SourceMemberContainerTypeSymbol type)
             {
-                this.NonTypeNonIndexerMembers = type.AddOrWrapTupleMembers(this.NonTypeNonIndexerMembers.ToImmutableAndFree());
+                this.NonTypeNonIndexerMembers = type.AddOrWrapTupleMembers(this.NonTypeNonIndexerMembers.ToImmutable());
             }
         }
 
-        protected virtual MembersAndInitializers? BuildMembersAndInitializers(DiagnosticBag diagnostics)
```

Comment:
>BuildMembersAndInitializers [](start = 50, length = 27)

Consider keeping this method and moving the work that ```protected MembersAndInitializers GetMembersAndInitializers()``` does to build final ```MembersAndInitializers``` here. Then, we don't need to make any changes to ```SimpleProgramNamedTypeSymbol``` and don't need to make ```BuildDeclaredMembersAndInitializers``` virtual.

Reply:
Especially that ```BuildDeclaredMembersAndInitializers``` cannot really return null unless it can check that private ```_lazyDeclaredMembersAndInitializers``` is not null.

---
In reply to: [562220687](https://github.com/dotnet/roslyn/pull/50650#discussion_r562220687) [](ancestors = 562220687)